### DISCUSSION TITLE
maintain `initialData` after receiving the results.

### DISCUSSION THREAD

**Discussion by vjuux** [TanStack/query#2427] [Q&A] [answered]
Most likely I'm misusing the api, but I was looking for a declarative way to add and maintain initial data 👀

**Comment by TkDodo** [comment, marked as answer]
Not sure what you mean by that. Do you have an example of what you want to achieve?

**Reply by vjuux** [reply]
Sure, when providing `initialData` as `{ initialData: [{ ID: 'Initial', name: 'Initial name' }] }` I would like to keep it as part of the final result after the data is fetched. Returned data would be something like 
`[{ ID: 'Initial', name: 'Initial name' }, { ID: 123, name: 'Some name' }, { ID: 124, name: 'etc etc' }]` instead of `[{ ID: 123, name: 'Some name' }, { ID: 124, name: 'etc etc' }]`

**Reply by TkDodo** [reply]
Then you need to merge it yourself. That is not what initialData does (emphasis on initial). Results from the queryFn could be an array or even a number. We can’t just merge those :)

**Reply by vjuux** [reply]
Fair enough, that's what I ended up doing. Thank you for your hard work and quick replies :)
